src/Nethermind/Nethermind.TxPool/Filters/GapNonceFilter.cs

```diff
@@ -25,12 +25,13 @@ public GapNonceFilter(TxDistinctSortedPool txs, ILogger logger)
 
         public AcceptTxResult Accept(Transaction tx, TxFilteringState state, TxHandlingOptions handlingOptions)
         {
+            bool isNotLocal = (handlingOptions & TxHandlingOptions.PersistentBroadcast) != TxHandlingOptions.PersistentBroadcast;
             int numberOfSenderTxsInPending = _txs.GetBucketCount(tx.SenderAddress!); // since unknownSenderFilter will run before this one
             bool isTxPoolFull = _txs.IsFull();
             UInt256 currentNonce = state.SenderAccount.Nonce;
             long nextNonceInOrder = (long)currentNonce + numberOfSenderTxsInPending;
             bool isTxNonceNextInOrder = tx.Nonce <= nextNonceInOrder;
-            if (isTxPoolFull && !isTxNonceNextInOrder)
+            if (isNotLocal && isTxPoolFull && !isTxNonceNextInOrder)
```

Comment:
So, we only filter it if tx pool is full? is that expected?

Reply:
🤔 Didn't notice, looks weird. But I think if we have space we can insert it. Why not? @marcindsobczak

Reply:
Yes, we don't mind to add gap-nonce txs to non-full TxPool - tx with missing nonce can come later. Nonce-gap tx's gas bootleneck will be 0 so they will be first to evict if our TxPool become full